src/useFormo.ts

```diff
@@ -522,144 +515,187 @@ export function useFormo<
-              });
-              return a;
-            }
-          )
-        )
-      ),
-      option.getOrElseW(() =>
-        taskEither.rightIO<NonEmptyArray<FieldError>, Values[K]>(() => {
-          setErrors(name, option.none);
+  ): Promise<Result<NonEmptyArray<FieldError>, Values[K]>> {
+    if (
+      fieldArrayValidators != null &&
+      fieldArrayValidators(values, index) != null &&
+      fieldArrayValidators(values, index)[name] != null &&
+      fieldArrayValidators(values, index)[name]![subfieldName] != null
+    ) {
+      const subfieldValidation = fieldArrayValidators(values, index)[name]![
+        subfieldName
+      ]!;
+      const result = (await subfieldValidation(
+        (values as V)[name][index][subfieldName] as any
+      )) as Result<NonEmptyArray<FieldError>, Values[K]>;
+      return matchResult(result, {
+        failure: (e) => {
           dispatch({
             type: "setFieldArrayErrors",
             field: name,
             index,
             subfield: subfieldName,
-            errors: option.none,
+            errors: e,
           });
-          return values[name];
-        })
-      )
-    );
+          return failure(e);
+        },
+        success: (a) => {
+          dispatch({
+            type: "setFieldArrayErrors",
+            field: name,
+            index,
+            subfield: subfieldName,
+            errors: undefined,
+          });
+          return success(a);
+        },
+      });
+    } else {
+      setErrors(name, undefined);
+      dispatch({
+        type: "setFieldArrayErrors",
+        field: name,
+        index,
+        subfield: subfieldName,
+        errors: undefined,
+      });
+      return success(values[name]);
+    }
+  }
 
-  const validateSubform = <
+  async function validateSubform<
     SK extends keyof ArrayRecordValues<Values> & string,
     K extends ArrayRecordKeys<Values>,
     V extends ArrayRecord<Values, SK>
   >(
     values: V,
     index: number,
     name: K
-  ): TaskEither<
-    NonEmptyArray<FieldError>,
-    ValidatedValues<
-      ArrayRecord<Values, keyof ArrayRecordValues<Values> & string>[K][number],
-      GetOrElse<ArrayValidators[K], {}>,
-      {}
+  ): Promise<
+    Result<
+      NonEmptyArray<FieldError>,
+      ValidatedValues<
+        ArrayRecord<
+          Values,
+          keyof ArrayRecordValues<Values> & string
+        >[K][number],
+        GetOrElse<ArrayValidators[K], {}>,
+        {}
+      >
     >
-  > => {
-    return record.traverseWithIndex(
-      taskEither.getTaskValidation(nonEmptyArray.getSemigroup<FieldError>())
-    )((subfieldName) => validateSubfield(name, index, subfieldName, values))(
-      values[name][index]
-    ) as any;
-  };
+  > {
+    let failures: Array<FieldError> = [];
+    let validatedValues = mapRecord(values[name][index], (v) => v);
+    for (let subfieldName in values[name][index]) {
+      const result = await validateSubfield(name, index, subfieldName, values);
```

Comment:
could/should be a `Promise.all` instead of loop, so as to run multiple async validations in parallel

Reply:
I agree it would be better to use `Promise.all` (or `Promise.allSettled` in case it helps).

Do you see any drawbacks?

Reply:
done in the last commit, it would probably need some new tests since the existing ones didn't fail 🤔